src/client/cli/cmd/snapshot.cpp

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (C) Canonical, Ltd.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 3.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "snapshot.h"
+
+#include "animated_spinner.h"
+#include "common_callbacks.h"
+#include "common_cli.h"
+
+#include <multipass/cli/argparser.h>
+
+namespace mp = multipass;
+namespace cmd = mp::cmd;
+
+mp::ReturnCode cmd::Snapshot::run(mp::ArgParser* parser)
+{
+    if (auto ret = parse_args(parser); ret != ParseCode::Ok)
+        return parser->returnCodeFrom(ret);
+
+    AnimatedSpinner spinner{cout};
+
+    auto on_success = [this, &spinner](mp::SnapshotReply& reply) {
+        spinner.stop();
+        fmt::print(cout, "Snapshot taken: {}.{}\n", request.instance(), reply.snapshot());
+        return ReturnCode::Ok;
+    };
+
+    auto on_failure = [this, &spinner](grpc::Status& status) {
+        spinner.stop();
+        return standard_failure_handler_for(name(), cerr, status);
+    };
+
+    spinner.start("Taking snapshot ");
```

Comment:
```suggestion
    spinner.start("Taking snapshot");
```

Hey @ricab, can we get rid of this extra space? Having the double space in the message (eg. `Taking snapshot  /` triggers my OCD.

Reply:
Oh sure, I hadn't noticed that.